fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -814,12 +845,30 @@ default boolean recordExists(@Nonnull final Tuple primaryKey, @Nonnull final Iso
         return recordExistsAsync(primaryKey, isolationLevel);
     }
 
+    /**
+     * Using the given index entry, resolve the primary key and asynchronously return the referenced record.
+     * @param index the index being scanned
+     * @param entry the index entry to be resolved
+     * @param orphanBehavior the {@link IndexOrphanBehavior} to apply if the record is not found
+     * @return the record referred to by the given index entry
+     * @deprecated in favor of {@link #loadIndexEntryRecord(Index, IndexEntry, IndexOrphanBehavior, ExecuteState)}
```

Comment:
I'm not sure how I feel about this being deprecated. It kind of seems like if there is a legitimate use case for a (non-`INTERNAL`) user loading a record from the index entry then letting them load without an `ExecuteState`. Unless the thinking is that this should really be `INTERNAL`.